### PR TITLE
Add `MiniscopeImagingExtractor`

### PR DESCRIPTION
**Pull request by weiglszonja**
Add extractor for the Miniscope format:

- `MiniscopeImagingExtractor` takes a `folder_path` that points to the main folder where the microscope and behavior video subfolders are located.

The expected folder structure for Miniscope V3 recordings, based on the example session received from the Tye lab, is organized as follows:

- `C6-J588_Disc5/` (main folder)
  - `15_03_28/` (subfolder corresponding to the recording time)
    - `Miniscope/` (subfolder containing the microscope video stream)
         - `0.avi` (file containing the microscope video)
         - `metaData.json` (file containing the metadata for the microscope device)

### CODE REVIEW

File: src/roiextractors/extractors/miniscopeimagingextractor/miniscopeimagingextractor.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
+import json
+import re
+from pathlib import Path
+from typing import Optional, Tuple, List
+
+import numpy as np
+from natsort import natsorted
+
+from neuroconv.datainterfaces.behavior.video.video_utils import VideoCaptureContext
```
Comment (CodyCBakerPhD) [inline src/roiextractors/extractors/miniscopeimagingextractor/miniscopeimagingextractor.py:9]: Should be moved inside a local scope if that's the approach we're taking

File: requirements-full.txt [hunk 2]
```diff
@@ -1,2 +1,3 @@
 tifffile>=2018.10.18
 scanimage-tiff-reader==1.4.1
+neuroconv[video]
```
Comment (CodyCBakerPhD) [inline requirements-full.txt:3]: Looks like you'll have to add `natsort` as well

**Comment by CodyCBakerPhD** [comment]
Mostly looks good; is there an example file to share yet?

File: requirements-full.txt (hunk 2 above)
Comment (weiglszonja) [reply requirements-full.txt:3]: > Looks like you'll have to add `natsort` as well

You're right, thanks!

> Mostly looks good; is there an example file to share yet?

I've just uploaded at `tye-lab-to-nwb/C6-J588_Disc5_miniscope_stub.nwb`

File: src/roiextractors/extractors/miniscopeimagingextractor/miniscopeimagingextractor.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+        """
+        if channel != 0:
+            raise NotImplementedError(
+                f"The {self.extractor_name}Extractor does not currently support multiple color channels."
+            )
+
+        end_frame = end_frame or self.get_num_frames()
+        start_frame = start_frame or 0
+
+        video = np.empty(shape=(end_frame - start_frame, *self.get_image_size()))
```
Comment (CodyCBakerPhD) [inline src/roiextractors/extractors/miniscopeimagingextractor/miniscopeimagingextractor.py:108]: @weiglszonja I think you might need to set `dtype="int16"` here; the file shows the data ending up in `float64`

I'd also ensure that the type returned by `cv2.cvtColor`  with `cv2.COLOR_RGB2GRAY` is `int16` not `float64` (and manually cast if so)

Comment (weiglszonja) [reply src/roiextractors/extractors/miniscopeimagingextractor/miniscopeimagingextractor.py:108]: `float64` is definitely incorrect, thank you for catching it! 

> I'd also ensure that the type returned by cv2.cvtColor with cv2.COLOR_RGB2GRAY is int16 not float64 (and manually cast if so)

The return type is actually `uint8` and not int16 which is the same as the `VideoCaptureContext.get_video_frame_dtype()` returns

Comment (CodyCBakerPhD) [reply src/roiextractors/extractors/miniscopeimagingextractor/miniscopeimagingextractor.py]: From meeting: yes, I suppose it does. OK then

**Comment by CodyCBakerPhD** [comment]
From meeting: dropping 3.7 here

@weiglszonja Can you try adding 3.11 in a follow-up just to see if it works?

**Comment by weiglszonja** [comment]
@CodyCBakerPhD Thanks for checking the file, turned out I was using an roiextractors version the commit before I fixed the[ data type issue](https://github.com/catalystneuro/roiextractors/pull/225/commits/30b3ea1fddc68c8efed10abd81d8ab28ed941e63). 
The example file has been replaced, now it is showing up as `uint8` values.

File: .github/workflows/testing.yml [hunk 4]
```diff
@@ -11,7 +11,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: [3.7, 3.8, 3.9]
+        python-version: [3.8, 3.9]
```
Comment (CodyCBakerPhD) [inline .github/workflows/testing.yml:14]:
```suggestion
        python-version: [3.8, 3.9, 3.10, 3.11]
```

Extending testing versions

File: .github/workflows/testing.yml [hunk 5]
```diff
@@ -11,7 +11,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: [3.7, 3.8, 3.9]
+        python-version: [3.8, 3.9, 3.10, 3.11]
```
Comment (CodyCBakerPhD) [inline .github/workflows/testing.yml:14]:
```suggestion
        python-version: [3.8, 3.9, "3.10", 3.11]
```

**Review by CodyCBakerPhD — APPROVED** [review]

**Comment by CodyCBakerPhD** [comment]
Awesome! Thanks for all the work on this

**Pull request merged**